fork-off/src/main.rs

```diff
@@ -85,28 +84,104 @@ async fn main() -> anyhow::Result<()> {
     Ok(())
 }
 
-async fn get_chain_state(http_rpc_endpoint: String) -> Vec<Value> {
-    let storage: Value = reqwest::Client::new()
-        .post(http_rpc_endpoint)
-        .json(&serde_json::json!({
+async fn get_key(http_rpc_endpoint: &str, key: &str, start_key: Option<&str>) -> Option<String> {
+    let body = match start_key {
+        Some(start_key) => {
+            serde_json::json!({
+                "jsonrpc": "2.0",
+                "id": 1,
+                "method": "state_getKeysPaged",
+                "params": [ key, 1, start_key ]
+            })
+        }
+        None => serde_json::json!({
             "jsonrpc": "2.0",
             "id": 1,
-            "method": "state_getPairs",
-            "params": ["0x"]
-        }))
+            "method": "state_getKeysPaged",
+            "params": [ key, 1 ]
+        }),
+    };
+
+    let response: Value = reqwest::Client::new()
+        .post(http_rpc_endpoint)
+        .json(&body)
         .send()
         .await
         .expect("Storage request has failed")
         .json()
         .await
         .expect("Could not deserialize response as JSON");
 
-    storage["result"]
+    debug!("get_key response: {}", response);
+
+    let result = response["result"]
         .as_array()
         .expect("No result in response")
+        .to_owned();
```

Comment:
We can save an allocation by dropping this `to_owned()`.

Reply:
done